.github/workflows/test.yml

```
@@ -0,0 +1,57 @@
+          ruff check leeq/
+          mypy leeq/ --ignore-missing-imports
+          
+      - name: Run tests with coverage
+        run: |
+          pytest tests/ -v --cov=leeq --cov-report=xml --cov-report=term \
+            --ignore=tests/theory/simulation/ \
+            --ignore=tests/utils/ai/ \
+            --ignore=tests/agents/
+          
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)